escape_roomba/thread_channel.py

```diff
@@ -368,7 +368,7 @@ async def _async_origin_updated(self, origin, rx_users):
                               manage_messages=True, manage_permissions=True)
 
         old = self.thread_channel.overwrites
-        old_overwrites = {u: o for u, o in old.items() if not o.is_empty}
+        old_overwrites = {u: o for u, o in old.items() if not o.is_empty()}
```

Comment:
Seems like this could have been caught by a test... 😉

Reply:
The effect was that it was including everything (empty or not) which meant it would tend to generate false positives in the comparison and update things unnecessarily... which is inefficient but still works, so tends to pass tests (and work in production).

It might be worth adding a test to verify that a no-change update doesn't refresh anything, which might catch many similar issues? Will plan to do that in a followup.